plugins/optimization-detective/site-health.php

```diff
@@ -0,0 +1,171 @@
+ *
+ * @since n.e.x.t
+ *
+ * @return array{label: string, status: string, badge: array{label: string, color: string}, description: string, actions: string, test: string} Result.
+ */
+function od_optimization_detective_rest_api_test(): array {
+	$result = array(
+		'label'       => __( 'The REST API endpoint is functional.', 'optimization-detective' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => __( 'Optimization Detective', 'optimization-detective' ),
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+			'<p>%s</p>',
+			__( 'Your site can send and receive URL metrics via the REST API endpoint.', 'optimization-detective' )
+		),
+		'actions'     => '',
+		'test'        => 'optimization_detective_rest_api',
+	);
+
+	$rest_url = get_rest_url( null, OD_REST_API_NAMESPACE . OD_URL_METRICS_ROUTE );
+	$response = wp_remote_post(
+		$rest_url,
+		array(
+			'headers'   => array( 'Content-Type' => 'application/json' ),
+			'sslverify' => false,
+		)
+	);
+
+	if ( is_wp_error( $response ) ) {
+		$result['status']      = 'recommended';
+		$result['label']       = __( 'Error accessing the REST API endpoint', 'optimization-detective' );
+		$result['description'] = esc_html__( 'There was an issue reaching the REST API endpoint. This might be due to server settings or the REST API being disabled.', 'optimization-detective' );
+		$info                  = array(
+			'error_message' => $result['description'],
+			'error_code'    => $response->get_error_code(),
+			'available'     => false,
+		);
+	} else {
+		$status_code     = wp_remote_retrieve_response_code( $response );
+		$data            = json_decode( wp_remote_retrieve_body( $response ), true );
+		$expected_params = array( 'slug', 'current_etag', 'hmac', 'url', 'viewport', 'elements' );
+		$info            = array(
+			'status_code' => $status_code,
+			'available'   => false,
+		);
+
+		if (
+			400 === $status_code
+			&& isset( $data['data']['params'] )
+			&& is_array( $data['data']['params'] )
+			&& count( $expected_params ) === count( array_intersect( $data['data']['params'], $expected_params ) )
+		) {
+			// The REST API endpoint is available.
+			$info['available'] = true;
+		} elseif ( 401 === $status_code ) {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'Authorization should not be required to access the REST API endpoint.', 'optimization-detective' );
+			$result['description'] = esc_html__( 'To collect URL metrics, the REST API endpoint should be accessible without requiring authorization.', 'optimization-detective' );
+		} elseif ( 403 === $status_code ) {
+			$result['status']      = 'recommended';
+			$result['label']       = __( 'The REST API endpoint should not be forbidden.', 'optimization-detective' );
+			$result['description'] = esc_html__( 'The REST API endpoint is blocked. Please review your server or security settings.', 'optimization-detective' );
```

Comment:
I think the messaging here should be updated. 403s often don't happen because the REST API endpoint is blocked, but because the user does not have the necessary capabilities.